mlrun/datastore/base.py

```diff
@@ -320,31 +324,17 @@ def reader(*args, **kwargs):
             raise Exception(f"File type unhandled {url}")
 
         if file_system:
-            if (
-                self.supports_isdir()
-                and file_system.isdir(file_url)
-                or self._is_dd(df_module)
-            ):
-                storage_options = self.get_storage_options()
-                if url.startswith("ds://"):
-                    parsed_url = urllib.parse.urlparse(url)
-                    url = parsed_url.path
-                    if self.using_bucket:
-                        url = url[1:]
-                    # Pass the underlying file system
-                    kwargs["filesystem"] = file_system
-                elif storage_options:
-                    kwargs["storage_options"] = storage_options
-                df = reader(url, **kwargs)
-            else:
-                file = url
-                # Workaround for ARROW-12472 affecting pyarrow 3.x and 4.x.
-                if file_system.protocol != "file":
-                    # If not dir, use file_system.open() to avoid regression when pandas < 1.2 and does not
-                    # support the storage_options parameter.
-                    file = file_system.open(url)
-
-                df = reader(file, **kwargs)
+            storage_options = self.get_storage_options()
```

Comment:
no special care for files anymore?

Reply:
Nope, the whole case [was added for an issue with pandas<1.2](https://github.com/mlrun/mlrun/pull/1049/files#diff-e6c49fcaee9a2be90cbb9d67f1efa5c48fa7e9bd008732d4cad59ee6a57d9ccaR213), which we no longer support.